Wikipedia/Code/NSUserActivity+WMFExtensions.m

```diff
@@ -241,7 +241,22 @@ - (WMFUserActivityType)wmf_type {
         return WMFUserActivityTypeSearchResults;
     } else {
         if ([self wmf_articleURL].wmf_isWikiResource) {
-            return WMFUserActivityTypeArticle;
+            if ([[[self wmf_articleURL] path] containsString:@":"]) {
+                __block WMFUserActivityType type = WMFUserActivityTypeArticle;
+                WMFTaskGroup *group = [WMFTaskGroup new];
+                [group enter];
+                [SessionSingleton.sharedInstance.dataStore.pageInfoFetcher fetchNamespaceValueForArticleWith:[self wmf_articleURL]
+                                                                                                  completion:^(NSNumber *_Nullable ns) {
+                                                                                                      if (ns.integerValue == PageNamespaceUserTalk) {
+                                                                                                          type = WMFUserActivityTypeUserTalk;
+                                                                                                      }
+                                                                                                      [group leave];
+                                                                                                  }];
+                [group wait];
```

Comment:
This `wait` will block the main thread for a network request when this method is called in  `canProcessUserActivity`. This can result in the [app being terminated if the network call takes too long](https://developer.apple.com/library/archive/qa/qa1693/_index.html). One alternative option would be to have AppVC make this call asynchronously while keeping the splash screen up or showing some other progress indicator. Another option would be to push the ArticleVC with this URL and have it display the talk page as a child view controller when the summary response comes back with namespace == 3 (similar to how external URLs are handled in `- (void)showExternalURL:(NSURL *)externalURL {`)

Reply:
If AppVC is making an asynchronous call, wouldn't `- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray<id<UIUserActivityRestoring>> *__nullable restorableObjects))restorationHandler` still have to hang on the main thread to know what to return?

Reply:
It can return true there like it did before based on `wmf_isWikiResource` then figure out whether it's a talk page asynchronously